sweat/algorithms/streams.py

```diff
@@ -0,0 +1,203 @@
+"""Operation on Streams that leave the shape of the stream unchanged"""
+
+import numpy as np
+import pandas as pd
+from sweat.utils import cast_array_to_original_type
+
+
+# FTP based 7-zones with left bind edge set to -0.001
+POWER_ZONES_THRESHOLD = [-0.001, 0.55, 0.75, 0.9, 1.05, 1.2, 1.5, 10.0]
+POWER_ZONES_THRESHOLD_DESC = ["Active Recovery", "Endurance", "Tempo",
+                              "Threshold", "VO2Max", "Anaerobic", "Neuromuscular",]
+POWER_ZONES_THRESHOLD_ZNAME = ["Z1", "Z2", "Z3", "Z4", "Z5", "Z6", "Z7"]
+
+# LTHR based 5-zones with left bind edge set to -0.001
+HEART_RATE_ZONES = [-0.001, 0.68, 0.83, 0.94, 1.05, 10.0]
+HEART_RATE_ZONES_DESC = ["Active recovery", "Endurance", "Tempo", "Threshold", "VO2Max",]
+HEART_RATE_ZONES_ZNAME = ["Z1", "Z2", "Z3", "Z4", "Z5"]
+
+
+def compute_zones(arg, **kwargs):
+    """Convert stream into respective zones stream
+
+    Watts streams can be converted either into ftp based 7-zones or into custom zones
+    HR streams can be converted either in lthr based 5-zones or into custom zones
+    One of three *ftp*, *lthr* or *zone* keyword parameters must be provided
+
+    Parameters
+    ----------
+    arg : array-like
+    ftp : number, optional
+        Value for FTP, will be used for 7-zones calculation
+    lthr: number, optional
+        Value for LTHR, will be used for 5-zones calculation
+    zones: list, optional
+        List of custom defined zones with left edge set to -1 and right edge to 10000
+
+    Returns
+    -------
+    array-like of int, the same type as arg
+    """
+
+    arg_s = pd.Series(arg)
+
+    if kwargs.get('zones', None):
+        abs_zones = kwargs.get('zones')
+
+    elif kwargs.get('ftp', None):
+        abs_zones = np.asarray(POWER_ZONES_THRESHOLD) * kwargs.get('ftp')
+
+    elif kwargs.get('lthr', None):
+        abs_zones = np.asarray(HEART_RATE_ZONES) * kwargs.get('lthr')
+
+    else:
+        raise ValueError
+
+    labels = kwargs.get('labels', list(range(1, len(abs_zones))))
+    assert len(abs_zones) == (len(labels) + 1)
+
+    y = pd.cut(arg_s, bins=abs_zones, labels=labels)
+    y = cast_array_to_original_type(y, type(arg))
+
+    return y
+
+
+
+def wpk(power, weight):
```

Comment:
I'm not sure if this streams file makes sense to me. For example for this wpk method it would make sense for me to live in the metrics module. What do you think?

Reply:
Agree.